datafusion/physical-expr/src/aggregate/sum.rs

```diff
@@ -297,6 +297,18 @@ pub(crate) fn sum(lhs: &ScalarValue, rhs: &ScalarValue) -> Result<ScalarValue> {
         (ScalarValue::Int64(lhs), ScalarValue::Int8(rhs)) => {
             typed_sum!(lhs, rhs, Int64, i64)
         }
+        (ScalarValue::Int64(lhs), ScalarValue::UInt64(rhs)) => {
```

Comment:
This looks like a fine change in this PR -- though it is strange to me that we have to be doing these casts in sum.rs as it duplicates some non trivial amount of the logic in coercion -- maybe it would be possible to make this code cleaner / consolidate more of the coercion logic.

Again, no changes needed for this PR but I figured I would point it out while reading this

Reply:
Thanks @alamb 
I'd look into this coercion logic in some follow up pr

Reply:
Coercion u64 to i64 seems irrational to me. Why do we need this kind of coercion in sum distinct?

Reply:
I think @WinkerDu  is just following the same pre-existing pattern with this PR

I agree that the pre-existing pattern doesn't make sense to me (it should have already been done by the time the distinct aggregate code is being executed)